### PR TITLE
Split client server

### PR DESCRIPTION
**Pull request by mpace965**
Splits `src` directory into `client` and `server` folders. Also updates all relevant paths that are affected by this change.

Now, `views`, `components`, and `styling` live in `client`. The rest lives in `server`.

Also, remember that "magic" that we talked about with local variables? Turns out we can replicate that somewhat with `app.locals`. I added the `assetPrefix` local, which automatically gets passed to anything rendered with the `app.render` middleware. So we were actually able to move the asset path into the view, and just interpolate the `assetPrefix`.

### CODE REVIEW

File: scripts/server.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ n=0
 until [ $n -ge 5 ]
 do
    yarn run migrate && break
-   n=$[$n+1]
+   n=$(($n+1))
```
Comment (mpace965) [inline scripts/server.sh:7]: Random fix. Noticed that the above version wasn't actually incrementing the counter, resulting in infinite retries :scream:

File: src/client/views/home.hbs [hunk 2]
```diff
@@ -5,3 +5,4 @@
 <div>{{> ex}}</div>
 
 <div id='root'></div>
+<script src='{{assetPrefix}}/js/hello.js'></script>
```
Comment (mpace965) [inline src/client/views/home.hbs:8]: Moved the react component script tag directly under the div it mounts to.

File: src/client/views/layouts/main.hbs [hunk 3]
```diff
@@ -3,18 +3,17 @@
...
-    <link rel='stylesheet' href='{{stylePath}}'>
+    <link rel='stylesheet' href='{{assetPrefix}}/css/styling.css'>
 </head>
 <body>
   <div class='container'>
     <div class='row'>
       <div class='col-md-1'></div>
       <div class='col-md-10'>
-        {{{body}}}  
+        {{{body}}}
```
Comment (mattgfisch) [inline src/client/views/layouts/main.hbs:13]: Thanks for fixing the trailing space!

Comment (mpace965) [reply src/client/views/layouts/main.hbs:13]: Its actually an Atom thing. Does that on save :+1:

File: scripts/server.sh (hunk 1 above)
Comment (mattgfisch) [reply scripts/server.sh:7]: Solid catch

File: src/client/views/layouts/main.hbs (hunk 3 above)
Comment (mattgfisch) [reply src/client/views/layouts/main.hbs:13]: Even more reason for me to switch lol

File: src/client/views/home.hbs (hunk 2 above)
Comment (mattgfisch) [reply src/client/views/home.hbs:8]: Makes sense

**Comment by mattgfisch** [comment]
Pulled it down and it ran it locally. Everything checks out!

🚢

**Comment by mpace965** [comment]
I forgot to squash :/

**Comment by mattgfisch** [comment]
No worries! The startup's repo has a GitHub configuration where it squashes the commits when you merge it. I'm not sure if you'd be interested in that or if you'd prefer we keep it manual so we have more control

**Comment by mpace965** [comment]
Sounds like it may be a good idea!

**Pull request merged**